helper/testhelpers/certhelpers/cert_helpers.go

```diff
@@ -0,0 +1,258 @@
+package certhelpers
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+type CertBuilder struct {
+	tmpl       *x509.Certificate
+	parentTmpl *x509.Certificate
+
+	selfSign  bool
+	parentKey *rsa.PrivateKey
+
+	isCA bool
+}
+
+type CertOpt func(*CertBuilder) error
```

Comment:
Nit: I noticed that in most(?) of the `CertOpt` functions defined in this file all return `nil`. Are there other functions that we have in mind that might not? If we don't think that's likely, should we change the method signature here and remove the `error`?

Reply:
Oh... I didn't write this... I just moved it out of the mongo specific space. :)

Reply:
Ah ok, well then I suppose for consistency we'll leave as is.